lib/puppet/type/sensu_client_config.rb

```diff
@@ -45,14 +45,31 @@ def insync?(is)
     end
   end
 
-  newproperty(:bind) do
-    desc "The IP that client will bind to"
-    defaultto '127.0.0.1'
-  end
+  newproperty(:socket) do
+    desc "A set of attributes that configure the Sensu client socket."
+    include PuppetX::Sensu::ToType
+
+    def is_to_s(hash = @is)
+      hash.keys.sort.map {|key| "#{key} => #{hash[key]}"}.join(", ")
+    end
```

Comment:
This breaks the puppet run for everyone updateing from an old puppet module.  Because the @is is '' or nil.    Is there a reason why just using hte default to_s function of the has isn't enough?

For example when upgrading to this commit running puppet get this error:
` Error: Could not convert change 'socket' to string: undefined method `keys' for nil:NilClass `

When I change is_to_s to just return hash it then updates cleanly with following log line
`Notice: /Stage[main]/Sensu::Client::Config/Sensu_client_config[sensu1b-af93681f.coreeng.mlbam.dev.us-east-1.bamgrid.net]/socket: socket changed '' to {'bind' => '127.0.0.1', 'port' => '3030'}`